### DISCUSSION TITLE
How to pass a variable from the livewire class to te controller

### DISCUSSION THREAD

**Discussion by iislamtaleb** [livewire/livewire#8152] [1. Help]
here Im dispatching an event to the livewire class here:

```
reserveButton.addEventListener('click',()=>{
    $wire.dispatch('make-reservation',{time:time,date:date});

});
```
Next, Im listening for the event in the livewire class:

```
#[\Livewire\Attributes\On('make-reservation')]
public function saveReservation($time,$date)
{
    return redirect()->route('home.appointmentReservation')
    ->with('time', $time)
    ->with('selectedDate',$date);
}
```
And here Im retreiving the variabels in the controller:

```
public function registerAppointment(Request $request){
    // Retrieve the $time variable from the session
    $time = $request->session()->get('time');
    $selectedDate = $request->session()->get('selectedDate');
    dd($time);
```
Everything is good till now, but I have a problem, when the user is not auth he must login first to perfome this event so that's way I added the middlware 'auth' to the route:

`Route::get('/appointment-reservation', [HomeController::class, 'registerAppointment'])->name('home.appointmentReservation')->middleware(['auth']);`
The problem is when the user get redirect to the login page the $time and $date variable will be null in function, but if the user is already authenticated and made the event everything works fine.

I have tried to many ways to pass the variables with thier data but failed. If you did not undersand just comment to explain again. Thanks

**Comment by joshhanley** [comment]
@iislamtaleb I would probably just pass them along a querystring parameters to the new route. Then your login controller should also use `redirect()->intended(...);` to ensure it goes back to the previous URL with querystrings.
